### PR TITLE
Add prop-types for several components

### PR DESCRIPTION
**Pull request by YHhaoareyou**
Please check if I added those prop-types correctly.

### CODE REVIEW

File: client/src/js/containers/syllabus/AddedCourseListContainer.js [hunk 1]
```diff
@@ -64,3 +64,9 @@ const mapDispatchToProps = {
 export default connect(mapStateToProps, mapDispatchToProps)(
   AddedCourseListContainer
 );
+
+AddedCourseListContainer.propTypes = {
+  addedCourses: PropTypes.array.isRequired,
+  semesterKey: PropTypes.string.isRequired,
+  changeCoursesSortingOption: PropTypes.func.isRequired
+};
```
Comment (OscarWang114) [inline client/src/js/containers/syllabus/AddedCourseListContainer.js]: I didn't mention it in the example, but `mapStateToProps` should also be examined. Basically, `mapDispatchToProps` maps a **dispatch function** to **props**, while `mapStateToProps` maps a slice of the **redux state tree** to **props**. So both of them provide props for the component. If you're not familiar with redux yet, I highly recommend you to learn it with its creator Dan Abramov [here](https://egghead.io/courses/getting-started-with-redux) (it's free).
```javascript
const mapStateToProps = (state, ownProps) => {
  return {
    selectedSortingOption: getSortingOption(
      state.addedCourses[ownProps.semesterKey]
    )
  };
};
```
You should add `selectedSortingOption` in `propTypes` as well

**Comment by OscarWang114** [comment]
There are some changes, but overall well done! 👍

**Comment by YHhaoareyou** [comment]
Thank you so much! I've already solved the problems you mentioned in the comments. Please confirm them.

File: README.md [hunk 2]
```diff
@@ -72,7 +72,3 @@ Submit an issue or a pull request! :blush:
 <!-- ALL-CONTRIBUTORS-LIST:END -->
 
 This project follows the [all-contributors](https://github.com/kentcdodds/all-contributors) specification.
-
-## License
```
Comment (OscarWang114) [inline README.md]: You should not delete the License section.

File: client/src/js/containers/syllabus/AddedCourseListContainer.js (hunk 1 above)
Comment (OscarWang114) [reply client/src/js/containers/syllabus/AddedCourseListContainer.js]: `getSortingOption` is **not** a prop. It is a function imported from a reducer and **used to generate the value for `selectedSortingOption`**. `selectedSortingOption` is the prop. Delete `getSortingOption` in `propTypes`.
```javascript
//AddedCourseListContainer.js...
import { getSortingOption } from '../../reducers/addedSemesterCourses';
```

Comment (OscarWang114) [reply client/src/js/containers/syllabus/AddedCourseListContainer.js]: The type of `selectedSortingOption` is **not** `object`. Check out `reducers/addedSemesterCourses.js` and try to figure out what data type `getSortingOption` returns.

**Comment by OscarWang114** [comment]
@YHhaoareyou Thanks for the changes! Some of them are still incorrect, so please fix them.

**Comment by YHhaoareyou** [comment]
About.js, App.js, FetchedCourseListSearch.js, and FilterGroup.js seem to be updated by other contributors already, so I only modified CourseChunk.js and FilterButton.js. Please check my update, thank you!

**Comment by OscarWang114** [comment]
Thanks @YHhaoareyou! I've merged your PR. Well done :D

**Comment by OscarWang114** [comment]
@YHhaoareyou Thanks for your PR again. @all-contributors please add @YHhaoareyou for code.

**Pull request merged**